pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -590,6 +590,11 @@ func (cm *containerManagerImpl) Start(node *v1.Node,
 		}
 	}
 
+	// Ensure that reserved resoures are valid.
+	if err := cm.validateReservedResources(); err != nil {
```

Comment:
This is to ensure that `hugepages-1234` and `hugepages-1gi` resources used in the `--{system,kube}-reserved` flags are caught and crashes kubelet. W

We can do this in `cmd/kubelet/app/server.go` in theory (we need to know what sizes the kubelet supports), but that will require importing linux-related stuff, making it more more thight to linux. Any thoughts?

Reply:
If we remove this without any other modifications, `--system-reserved=hugepages-1Gi=2` on a node without support for 1Gi pages, will silently ignore it and just carry on. The same applies to typos like `hugepages-1gi`, `hugepages-1G` etc.

Reply:
I'm also curious about what the best behavior in this scenario is... looking forward to the community weighing in :)

Reply:
this is fine.  windows has largepages but its not really enabled in kube today.